plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -223,15 +223,22 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
     private fun processBackingFields(symbol: FirPropertySymbol, embedding: ClassTypeEmbedding): Pair<SimpleKotlinName, FieldEmbedding>? {
         val unscopedName = symbol.callableId.embedUnscopedPropertyName()
         // This field is already registered in the supertype: we don't need to know about it.
-        if (embedding.findAncestorField(unscopedName) != null) return null
+        val ancestorField = embedding.findAncestorField(unscopedName)
+        if (ancestorField != null) {
+            if (symbol.fromPrimaryConstructor)
+                ancestorField.addTypeContainingInPrimaryConstructor(embedding, symbol)
+            return null
+        }
         val name = symbol.callableId.embedMemberPropertyName()
 
         val backingField = name.specialEmbedding() ?: symbol.hasBackingField.ifTrue {
             UserFieldEmbedding(
                 name,
                 embedType(symbol.resolvedReturnType),
                 symbol
-            )
+            ).also {
+                if (symbol.fromPrimaryConstructor) it.addTypeContainingInPrimaryConstructor(embedding, symbol)
```

Comment:
This is duplicated with the code above, maybe it would be good to split it into a helper function?